packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -24,8 +29,67 @@ class Tooltip extends ElementMixin(PolymerElement) {
           display: none;
         }
       </style>
+      <vaadin-tooltip-overlay
+        id="[[_uniqueId]]"
+        role="tooltip"
+        theme$="[[_theme]]"
+        position-target="[[target]]"
+        modeless
+      ></vaadin-tooltip-overlay>
     `;
   }
+
+  static get properties() {
+    return {
+      /**
+       * The id of the element used as a tooltip trigger.
+       */
+      for: {
+        type: String,
+        observer: '__forChanged',
+      },
+
+      /**
+       * Reference to the element used as a tooltip trigger.
+       * The target must be placed in the same shadow scope.
+       * Defaults to an element referenced with `for`.
+       */
+      target: {
+        type: Object,
+        observer: '__targetChanged',
+      },
+    };
+  }
+
+  constructor() {
+    super();
+
+    this._uniqueId = `vaadin-tooltip-${generateUniqueId()}`;
+  }
+
+  /** @private */
+  __forChanged(forId) {
+    if (forId) {
+      const target = this.getRootNode().getElementById(forId);
```

Comment:
Thanks, this is a valid point. I don't see how we could fix this without adding MutationObserver to track additions to the parent scope, which is something I would like to avoid. Should we document the following limitation?

> the target element is in the DOM by the time this observer is run

Reply:
Good enough IMO 👍

Reply:
Added the note to the property JSDoc.